doc/source/whatsnew/v0.25.0.rst

```diff
@@ -322,7 +322,7 @@ Sparse
 Other
 ^^^^^
 
--
+- Bug in :class:`BaseExprVisitor` for using named kwargs in :func:`eval` expressions.
```

Comment:
Can you come up with a more user-facing message? 

Also if the focus is on passing keywords args through eval can you add a test which explicitly does this?

Reply:
I've updates the tests and whatsnew message